testng-core/src/main/java/org/testng/SuiteRunner.java

```diff
@@ -252,10 +252,14 @@ public void setReportResults(boolean reportResults) {
   }
 
   private void invokeListeners(boolean start) {
-    for (ISuiteListener sl : Lists.newArrayList(listeners.values())) {
-      if (start) {
+    if (start) {
+      for (ISuiteListener sl : Lists.newArrayList(listeners.values())) {
         sl.onStart(this);
-      } else {
+      }
+    } else {
+      List<ISuiteListener> suiteListenersReversed = new ArrayList<>(listeners.values());
+      Collections.reverse(suiteListenersReversed);
+      for (ISuiteListener sl : suiteListenersReversed) {
```

Comment:
Why not use the newly added method `Lists. newReversedArrayList()`

Reply:
Thank you for your reminder!  Fixed, pls have a look,